packages/app/src/app/pages/common/Modals/MidTrialModal/index.tsx

```diff
@@ -0,0 +1,301 @@
+
+  useEffect(() => {
+    experimentPromise.then(experiment => {
+      setExperimentValue(experiment);
+    });
+  }, [experimentPromise]);
+
+  const checkoutUrl = useGetCheckoutURL({
+    success_path: pathname,
+    cancel_path: pathname,
+  });
+
+  const handleClose = () => {
+    track('Mid trial reminder: close modal', {
+      codesandbox: 'V1',
+      event_source: 'UI',
+    });
+
+    dismissMidTrialReminder();
+    modalClosed();
+  };
+
+  const handleUpgrade = () => {
+    dismissMidTrialReminder();
+
+    track('Mid trial reminder: upgrade to pro', {
+      codesandbox: 'V1',
+      event_source: 'UI',
+    });
+  };
+
+  return (
+    <Stack css={{ padding: '24px' }} direction="vertical" gap={12}>
+      <Stack justify="flex-end">
+        <IconButton name="cross" title="Close" onClick={handleClose} />
+      </Stack>
+      <Stack direction="vertical" gap={6} css={{ textAlign: 'center' }}>
+        <Stack direction="vertical" gap={1}>
+          <Text
+            as="h2"
+            size={32}
+            color="#FFFFFF"
+            align="center"
+            css={{ marginTop: 0 }}
+            fontFamily="everett"
+          >
+            {remainingTrialDays === 7
+              ? 'One week'
+              : `${remainingTrialDays} days`}{' '}
+            left on your trial.
+          </Text>
+          <Text as="p" size={19} color="#C2C2C2" align="center">
+            {!experimentValue ? <SkeletonText /> : null}
+
+            {experimentValue === ExperimentValues.A
+              ? 'You will lose access to all Pro features when your trial expires.'
+              : null}
+            {experimentValue === ExperimentValues.B
+              ? 'You will need a Team Pro subscription to continue using Pro features.'
+              : null}
+          </Text>
+        </Stack>
+
+        {!experimentValue ? (
+          <Stack direction="vertical" gap={4}>
+            <SkeletonText />
+            <SkeletonText />
+            <SkeletonText />
+            <SkeletonText />
+            <SkeletonText />
+            <SkeletonText />
+            <SkeletonText />
+          </Stack>
+        ) : null}
+
+        {experimentValue === ExperimentValues.A ? <LoseAfterTrial /> : null}
+        {experimentValue === ExperimentValues.B ? <KeepWithTrial /> : null}
+      </Stack>
+      <Stack gap={4} justify="space-between">
+        <Element css={{ flexGrow: 1 }}>
+          <Button onClick={handleClose} variant="secondary">
+            Continue trial
+          </Button>
+        </Element>
+        <Element css={{ flexGrow: 1 }}>
+          <Button
+            as="a"
+            href={checkoutUrl}
+            onClick={handleUpgrade}
+            variant="primary"
+          >
+            Upgrade to pro
+          </Button>
+        </Element>
+      </Stack>
+    </Stack>
+  );
+};
+
+// Based on FEATURES in SubscriptionCancellation.tsx, they're just in a different
+// order.
```

Comment:
Would it be possible to reuse the constants from SubscriptionCancellation if the order was the same?